File: app/src/main/java/de/garten/training/depotflow/ui/LegacyUiFormatter.java
package de.garten.training.depotflow.ui;

import de.garten.training.depotflow.data.db.green.ChecklistItem;
import de.garten.training.depotflow.data.db.green.Stop;
import de.garten.training.depotflow.data.db.green.WorkOrder;

public class LegacyUiFormatter {

    public String formatOrderSubtitle(WorkOrder order) {
        return order.getCustomerName() + " · Priorität " + order.getPriority() + " · fällig " + nullSafe(order.getDueAt());
    }

    public String formatOrderStatus(WorkOrder order) {
        String sync = order.getSyncStatus() == null ? "unknown" : order.getSyncStatus().name();
        return "Status: " + order.getStatus() + " · Sync: " + sync + (order.isDirty() ? " · lokal geändert" : "");
    }

    public String formatStop(Stop stop) {
        return stop.getSequence() + ". " + stop.getType() + " · " + stop.getName() + "\n" +
                nullSafe(stop.getAddress()) + "\n" +
                "Fenster: " + nullSafe(stop.getArrivalWindowFrom()) + " – " + nullSafe(stop.getArrivalWindowTo());
    }

    public String formatChecklistItem(ChecklistItem item) {
        String marker = item.isChecked() ? "[x] " : "[ ] ";
        String mandatory = item.isMandatory() ? " (Pflicht)" : "";
        return marker + item.getLabel() + mandatory + optionalNote(item.getNote());
    }

    private String optionalNote(String note) {
        if (note == null || note.trim().isEmpty()) {
            return "";
        }
        return "\n    Notiz: " + note;
    }

    private String nullSafe(String value) {
        return value == null ? "-" : value;
    }
}

File: app/src/main/java/de/garten/training/depotflow/ui/LegacyUiFormatter.kt
package de.garten.training.depotflow.ui

import de.garten.training.depotflow.data.db.green.ChecklistItem
import de.garten.training.depotflow.data.db.green.Stop
import de.garten.training.depotflow.data.db.green.WorkOrder

class LegacyUiFormatter {
    fun formatOrderSubtitle(order: WorkOrder): String {
        return order.customerName + " · Priorität " + order.priority + " · fällig " + nullSafe(order.dueAt)
    }

    fun formatOrderStatus(order: WorkOrder): String {
        val sync = if (order.syncStatus == null) "unknown" else order.syncStatus!!.name
        return "Status: " + order.status + " · Sync: " + sync + (if (order.isDirty) " · lokal geändert" else "")
    }

    fun formatStop(stop: Stop): String {
        return stop.sequence.toString() + ". " + stop.type + " · " + stop.name + "\n" +
                nullSafe(stop.address) + "\n" +
                "Fenster: " + nullSafe(stop.arrivalWindowFrom) + " – " + nullSafe(stop.arrivalWindowTo)
    }

    fun formatChecklistItem(item: ChecklistItem): String {
        val marker = if (item.isChecked) "[x] " else "[ ] "
        val mandatory = if (item.isMandatory) " (Pflicht)" else ""
        return marker + item.label + mandatory + optionalNote(item.note)
    }

    private fun optionalNote(note: String?): String {
        if (note == null || note.trim { it <= ' ' }.isEmpty()) {
            return ""
        }
        return "\n    Notiz: " + note
    }

    private fun nullSafe(value: String?): String {
        return if (value == null) "-" else value
    }
}
